### ISSUE TITLE
Feature Request: Display captured information in Find in files/Find all in current document

### ISSUE THREAD

**Issue by Loceka** [notepad-plus-plus/notepad-plus-plus#11109]
When using Find in files and Find all in current document the result displays all and only the first line of the whole match.

In regular expression mode, it would be nice to display only the captured parts of the regular expression (and revert to the default mode if there are no captured parts).

Eg. the regex search `(m(?:in|ax)Amount:)(?:.*\n){5}\s+example:\s*(\d+)` over the text ```
          properties:
            minAmount: # minimum amount
              description: minimum amount constraint
              type: integer
              format: int64
              minimum: 0
              example: 500
            maxAmount: # maximum amount
              description: maximum amount constraint
              type: integer
              format: int64
              minimum: 0
              example: 200000
```

Current display:
![Screenshot_1 - Copie](https://user-images.githubusercontent.com/7732470/151761453-969009f6-9f68-450a-902b-cb0e941fb7a8.png)

Improved display:
![Screenshot_1](https://user-images.githubusercontent.com/7732470/151761486-3df21080-8ffd-49d0-875e-e22f288ca3d1.png)

**Comment by dfs-** [comment]
If I understand you correctly, you want to define a pattern for matching groups to be displayed in the search result window, but obviously only for regular expressions?

I don't know how user-friendly such a feature would be. I guess a RegEx-Tester like https://regex101.com/ or https://regexr.com/ might be a better way to do such things. Even PowerShell's "Select-String" method might be helpful.

**Comment by alankilborn** [comment]
@Loceka 

A more correct regular expression might be: `(?s)(m(?:in|ax)Amount:)(?:.*?\R){5}\s+example:\s*(\d+)`

**Comment by alankilborn** [comment]
I agree with @dfs- that this feature would not be generally useful.  Recommend closure of this issue.

**Comment by alankilborn** [comment]
@Loceka 

You can implement the desired functionality with a PythonScript; example:

    # -*- coding: utf-8 -*-
    from __future__ import print_function

    from Npp import *

    #-------------------------------------------------------------------------------

    class I11109(object):

        def __init__(self):
            matches = []
            self.accum = ''
            def match_found(m):
                for x in range(len(m.groups())):
                    self.accum += m.group(x + 1)
                self.accum += '\r\n'
            editor.research(r'(?s)(m(?:in|ax)Amount:)(?:.*?\R){5}\s+example:\s*(\d+)', match_found)
            if self.accum:
                notepad.new()
                editor.setText(self.accum)

    #-------------------------------------------------------------------------------

    if __name__ == '__main__': I11109()

    #-------------------------------------------------------------------------------

    '''
              properties:
                minAmount: # minimum amount
                  description: minimum amount constraint
                  type: integer
                  format: int64
                  minimum: 0
                  example: 500
                maxAmount: # maximum amount
                  description: maximum amount constraint
                  type: integer
                  format: int64
                  minimum: 0
                  example: 200000
    '''

**Comment by Loceka** [comment]
> If I understand you correctly, you want to define a pattern for matching groups to be displayed in the search result window, but obviously only for regular expressions?
> 
> I don't know how user-friendly such a feature would be. I guess a RegEx-Tester like https://regex101.com/ or https://regexr.com/ might be a better way to do such things. Even PowerShell's "Select-String" method might be helpful.

You don't understand the original need :
I have several files on the disk that contain the information I want and I would have found it nice to have the whole information directly in the search results instead of having to open each file to see the information myself (and here there were a lot of files).

Since the search result display for "find in files" and "find all in current document" does already display the line, it would just be nice to have it display the matched result instead, when a regex does have capture groups.

I guess it can also be useful to test regex but that was not at all the point here.

@alankilborn 
Where do you use this python script ? I thought N++ plugins where in C++

**Comment by alankilborn** [comment]
>Where do you use this python script ? I thought N++ plugins where in C++

_PythonScript_ is a plugin; such scripts run under its control.
